app/main.py

```diff
@@ -1,13 +1,20 @@
 def create_dict(*args):
     result_dict = {}
     for n, arg in enumerate(args):
-        if (not isinstance(arg, list) or not
-                isinstance(arg, set) or not
-                isinstance(arg, dict)):
-            try:
-                result_dict[arg] = n
-            except TypeError:
-                print(f"Cannot add {arg} to the dict!")
+        if isinstance(arg, tuple):
+            temp_list = []
+            for arg2 in arg:
+                if not isinstance(arg2, (list, set, dict)):
+                    temp_list.append(arg2)
+                    continue
+                else:
+                    print(f"Cannot add {arg} to the dict!")
+                    temp_list = []
+            if temp_list:
+                tuple_temp = tuple(temp_list)
```

Comment:
You can add not all elements from "arg" and it can be possible that "tuple(temp_list) != arg" and you add to dict wrong element, check this block